internal/telemetry/client.go

```diff
@@ -449,19 +515,44 @@ func (c *Client) backgroundHeartbeat() {
 	if !c.started {
 		return
 	}
-	err := c.submit(c.newRequest(RequestTypeAppHeartbeat))
-	if err != nil {
-		c.log("heartbeat failed: %s", err)
-	}
+	// TODO (evan.li): thoughts on calling go func here?
+	//				   don't want the request to block while holding the lock
+	c.flush()
```

Comment:
Yea that was the intent, though I agree enqueueing the heartbeat event is good for now and is more efficient. 

For future work the heartbeat event should be prioritized over other waiting events (e.g. we don't want a situation where the heartbeat event is the last in the queue and we start getting a `429` response error before we can send the heartbeat event). This is because other events can be saved for later to retry (using the new `extended-heartbeat` event which is not implemented yet), but the heartbeat event needs to be sent during every heartbeat interval. But this logic can be implemented in a separate PR that supports `extended-heartbeat`